verification/cbmc_verification_client.py

```diff
@@ -1,68 +1,126 @@
+            tmp_f.write(source_code_content)
+            tmp_f.seek(0)
+            path_to_file = Path(tmp_f.name)
+            current_context = self._context_manager.current_context(
+                function=function,
+                proof_state=proof_state,
+            )
+            vinput = VerificationInput(
+                function=function,
+                spec=spec,
+                context=current_context,
+                contents_of_file_to_verify=source_code_content,
+            )
+            if vinput not in self._cache:
+                logger.debug(f"vresult was not cached for: {vinput}")
+                vcommand = self._get_cbmc_verification_command(
+                    vinput, path_to_file_to_verify=path_to_file, proof_state=proof_state
+                )
+                try:
+                    logger.debug(f"Running command: {vcommand}")
+                    result = subprocess.run(vcommand, shell=True, capture_output=True, text=True)
+                    failure_messages = (
+                        None
+                        if result.returncode == 0
+                        else f"STDOUT: {result.stdout}\nSTDERR: {result.stderr}"
+                    )
+                    self._cache[vinput] = VerificationResult(
+                        vinput, succeeded=result.returncode == 0, failure_messages=failure_messages
+                    )
+                    logger.debug(f"Cached vresult for: {vinput}")
+                except Exception as e:
+                    msg = f"Error running command for function {function.name}: {e}"
+                    raise RuntimeError(msg) from e
+            vresult = self._cache[vinput]
+            if vresult.succeeded:
+                logger.success(f"Verification succeeded for function '{function.name}'")
+            else:
+                logger.error(f"Verification failed for function '{function.name}'")
+            return self._cache[vinput]
+
+    def _get_cbmc_verification_command(
+        self,
+        verification_input: VerificationInput,
+        path_to_file_to_verify: Path,
+        proof_state: ProofState,
+    ) -> str:
+        """Return the command used to verify a function in a file with CBMC.
+
+        Args:
+            verification_input (VerificationInput): The verification input.
+            path_to_file_to_verify (Path): The path to the file to verify.
+            proof_state (ProofState): The proof state to verify under.
 
         Returns:
-            Success | Failure: Success if the function successfully verifies, Failure if the
-                function does not verify.
+            str: The command used to verify a function in a file with CBMC.
         """
+        function_name = verification_input.function.name
+        # MDE: Do we want to replace *every* function (except the one being verified) by its current
+        # specification, whether or not it is verified/assumed?  Please discuss.
```

Comment:
@mernst Shouldn't we? If we don't replace each function with its spec, isn't there a possibility of generating a spec that might verify only in the absence of other specs (i.e., those of any callees?)